### PR TITLE
[tempo-distributed] add headless service for tempo-distributor

### PR DESCRIPTION
**Pull request by AlphaStyle**
This merge request will add a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) for the tempo-distributor.

When the OpenTelemetry Collector is using the [Load Balancing Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/loadbalancingexporter/README.md), and the load balancing exporter is configured to used the `dns` resolver, a headless service for the tempo-distributor is required for the dns resolver to get all endpoints. Without a headless service, the load balancing exporter will only see the Kubernetes service IP, and not the endpoints.

If you have suggestions, improvements or questions, feel free to ask or enlighten me.

### CODE REVIEW

**Comment by mapno** [comment]
You can also export via HTTP and send traces to [these endpoints](https://github.com/grafana/helm-charts/blob/main/charts/tempo-distributed/values.yaml#L1384-L1394), and the gateway will handle the load balancing. I'm ok with merging this as well if needed.

**Comment by AlphaStyle** [comment]
We just tried what you suggested. Setting up a `otelhttp` exporter in OpenTelemetry Collector and using the tempo-gateway as the endpoint does indeed work as you said.

The advantage with this Pull Request is that you can also load balance grpc. Just for the sake of alternatives, I would like to continue with the pull request :+1:

**Review by mapno — APPROVED** [review]
Agree, we can add this as an alternative.

File: charts/tempo-distributed/templates/distributor/service-distributor-discovery.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    - name: http-metrics
+      port: 3100
+      targetPort: http-metrics
+    - name: grpc
+      port: 9095
+      protocol: TCP
+      targetPort: 9095
+      {{- if .Values.distributor.appProtocol.grpc }}
+      appProtocol: {{ .Values.distributor.appProtocol.grpc }}
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (AlphaStyle) [reply charts/tempo-distributed/templates/distributor/service-distributor-discovery.yaml]: Did a rebase, and removed `9095` port from the headless distributor service :+1:

File: charts/tempo-distributed/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: tempo-distributed
 description: Grafana Tempo in MicroService mode
 type: application
-version: 0.27.19
+version: 0.27.20
```
Comment (zanhsieh) [inline charts/tempo-distributed/Chart.yaml:5]: Feature added should be bumped to minor version.
```suggestion
version: 0.28.0
```

File: charts/tempo-distributed/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # tempo-distributed
 
-![Version: 0.27.19](https://img.shields.io/badge/Version-0.27.19-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.5.0](https://img.shields.io/badge/AppVersion-1.5.0-informational?style=flat-square)
+![Version: 0.27.20](https://img.shields.io/badge/Version-0.27.20-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.5.0](https://img.shields.io/badge/AppVersion-1.5.0-informational?style=flat-square)
```
Comment (zanhsieh) [inline charts/tempo-distributed/README.md:3]:
```suggestion
![Version: 0.28.0](https://img.shields.io/badge/Version-0.28.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.5.0](https://img.shields.io/badge/AppVersion-1.5.0-informational?style=flat-square)
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**